### PR TITLE
66 add reference editor page

### PR DESCRIPTION
**Pull request by Shawk031**
# Summary
Adds reference page with Edit functionality. This is practically identical to the one existing in DCHP-2

TODO:
1. implement "add reference" functionality
2. implement "delete reference" functionality
3. improve CSS

### CODE REVIEW

File: app/routes/reference/$id.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  const data = Object.fromEntries(await request.formData())
+  invariant(data.id)
+  invariant(data.shortDisplay)
+  invariant(data.referenceText)
+
+  const id = getNumberFromFormInput(data.id)
+  const shortDisplay = getStringFromFormInput(data.shortDisplay)
+  const referenceText = getStringFromFormInput(data.referenceText)
+
+  console.log(id)
```
Comment (hangler) [inline app/routes/reference/$id.tsx:38]: No need to fix this now, but in the future let's clean up any `console.log`s. Yes I am guilty of this myself. Haha.

File: app/utils/generalUtils.ts [hunk 2]
```diff
@@ -48,6 +48,10 @@ export function validateEmail(email: unknown): email is string {
   return typeof email === "string" && email.length > 3 && email.includes("@")
 }
 
+export function stripHtml(text: string) {
+  return text.replace(/<\/?[^>]+(>|$)/g, "")
```
Comment (hangler) [inline app/utils/generalUtils.ts:52]: In general, I would prefer to use an existing library for this kind of thing, e.g. https://www.npmjs.com/package/string-strip-html (haven't looked into it at all but I assume it does what it says it does). Handling HTML tags in strings isn't actually all that straightforward. Here is a classic post on the topic: https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454

**Review by hangler — APPROVED** [review]

File: app/routes/reference/$id.tsx (hunk 1 above)
Comment (Shawk031) [reply app/routes/reference/$id.tsx:38]: LOL yup

File: app/utils/generalUtils.ts (hunk 2 above)
Comment (Shawk031) [reply app/utils/generalUtils.ts:52]: I made a ticket for this! Should be simple

**Pull request merged**